src/envoy/tcp/mixer/control_factory.h

```diff
@@ -62,14 +62,12 @@ class ControlFactory : public Logger::Loggable<Logger::Id::filter> {
 
   // The config object
   std::shared_ptr<Config> config_;
-  // The cluster manager
-  Upstream::ClusterManager& cm_;
   // the thread local slots
   ThreadLocal::SlotPtr tls_;
   // The statistics struct.
-  Utils::MixerFilterStats stats_;
+  std::shared_ptr<Utils::MixerFilterStats> stats_;
   // UUID of the Envoy TCP mixer filter.
-  const std::string uuid_;
+  std::shared_ptr<const std::string> uuid_;
```

Comment:
No,  shared_ptr for these two variables is over-kill.

I believe stats_ itself is already "&".     uuid is a string, we can copy that string to the capture.

Reply:
How about this idea:

Add a new class ControlData to hold all ControlFactory data except tls_

ControlFactory will hold 

tls_
and shared_ptr<ControlData>,

ControlDataSharedPtr will pass to Lambda to make sure

Reply:
Per discussion, I will create std::shared_ptr<ControlFactoryData>, and move cm_, stats_, uuid_, and config_ into ControlFactoryData. I will pass ControlFactoryData into capture list.

Reply:
Added ControlDataSharedPtr control_data_ into ControlFactory.